frontend/wallet/src/render/views/onboarding/InstallCodaStep.re

```diff
@@ -32,17 +32,27 @@ module Styles = {
 };
 [@bs.scope "window"] [@bs.val] external openExternal: string => unit = "";
 
+type state =
+  | Init
+  | Downloading
+  | Installing
+  | Finished;
+
 module InstallProgress = {
   [@react.component]
-  let make = (~setFinished, ~finished) =>
+  let make = (~setState, ~installerState) =>
     <div className=Styles.installer>
       <div className=Styles.downloader>
         <Downloader
-          keyName="keys-temporary_hack-testnet_postake.tar.bz2"
-          onFinish={_ => setFinished(_ => true)}
-          finished
+          onFinish={result =>
+            switch (result) {
+            | Belt.Result.Ok(_) => setState(_ => Finished)
+            | Belt.Result.Error(_) => setState(_ => Finished)
+            }
```

Comment:
This switch doesn't seem to do much

Reply:
Setting up for more granular (better error handling, etc) states in a follow on PR.